src/istgt_lu_ctl.c

```diff
@@ -2902,6 +2903,88 @@ _verb_istat ISCSIstat_last[ISCSI_ARYSZ] = { {0,0,0} };
 _verb_istat ISCSIstat_now[ISCSI_ARYSZ] = { {0,0,0} };
 _verb_istat ISCSIstat_rslt[ISCSI_ARYSZ] = { {0,0,0} };
 
+// istgt_uctl_cmd_iostats collects the iostats from the spec structure
+// and marshal them into json format using json-c library.The returned
+// string memory is managed by the json_object and will be freed when
+// the reference count of the json_object drops to zero.
+// TODO: Add the fields for getting the latency, used capacity etc.
+
+static int
+istgt_uctl_cmd_iostats(UCTL_Ptr uctl)
+{
+	ISTGT_LU_Ptr lu;
+	int rc, length;
+	// instantiate json_object from json-c library.
+        struct json_object *jobj;
+	// these are utility variables that will be freed at the end of the
+	// function.
+	char *writes, *reads, *totalreadbytes, *totalwritebytes, *size;
+	int i, j;
+	ISTGT_LU_DISK *spec;
+	ISTGT_Ptr istgt = uctl->istgt;
+	for (i = 0; i < MAX_LOGICAL_UNIT; i++) {
+                lu = (ISTGT_LU_Ptr *) istgt->logical_unit[i];
+                if (lu == NULL)
```

Comment:
instead of traversing through istgt->logical_unit, we can traverse spec_q to avoid nested looping.